### PR TITLE
Add Dockerfile for 20 al2023

### PR DESCRIPTION
**Pull request by rgithubli**
### Notes
https://github.com/corretto/corretto-docker/issues/164

Providing corretto20 images for AL2023. 

### Callouts

Corretto 20 doesn't have AL2023 rpm, so we installed generic linux on AL2023. In order to make the basic jdk to be functional, also installed some libs (`fontconfig`, `freetype` etc.) However, it's still possible that users need to install required libs if exceptions happen.

Had to install gnupg2 (the preinstalled `gnupg2-minimal` doesn't provide import function). However can't uninstall it - the dnf would complain `The operation would result in removing the following protected packages: dnf`. Just leave it there.

### Tests

Can run a basic HelloWorld in the container

### CODE REVIEW

**Comment by lutkerd** [comment]
Don't you need to update bin/update-dockerfiles.sh, bin/test-image.sh and bin/tag-generator.py to this as well?

File: 20/jdk/al2023/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    && dnf install -y java-20-amazon-corretto-devel-$version \
+    && (find /usr/lib/jvm/java-20-amazon-corretto -name src.zip -delete || true) \
+    && dnf remove -y findutils \
+    && dnf install -y fontconfig \
+    && dnf install -y freetype \
+    && dnf install -y libjpeg \
+    && dnf install -y dejavu-sans-fonts \
+    && dnf install -y dejavu-serif-fonts \
+    && dnf install -y dejavu-sans-mono-fonts \
+    && dnf install -y libjpeg \
```
Comment (lutkerd) [inline 20/jdk/al2023/Dockerfile:31]: You can install all of these in 1 command.

**Comment by rgithubli** [comment]
> Don't you need to update bin/update-dockerfiles.sh, bin/test-image.sh, bin/tag-generator.py and .github/workflows/verify-images.yml to this as well?

Yes. Didn't know they are in the package. Will do in the next rev.

File: 20/jdk/al2023/Dockerfile (hunk 1 above)
Comment (rgithubli) [reply 20/jdk/al2023/Dockerfile:31]: Will do

File: .tags [hunk 2]
```diff
@@ -0,0 +1,159 @@
+Tags: 8, 8u382, 8u382-al2, 8-al2-full, 8-al2-jdk, 8-al2-generic, 8u382-al2-generic, 8-al2-generic-jdk, latest
```
[the remark this answers is not in the document]
Comment (lutkerd) [reply .tags:1]: I don't think this file actually matters, we pull it from the auto-gen comment on the commits.

File: bin/tag-generator.py [hunk 3]
```diff
@@ -4,13 +4,15 @@
...
 LTS_VERSIONS = [ "8", "11", "17"]
+AL2023_GENERIC_VERSIONS = ["20"]
 
 def generate_tags(key, version):
     update = version.split('.')[1] if (key == '8') else version.split('.')[2]
     expanded_version = f"{key}u{update}" if (key == '8') else f"{key}.0.{update}"
 
     al2_tags = [f"{key}", f"{expanded_version}", f"{expanded_version}-al2", f"{key}-al2-full", f"{key}-al2-jdk", f"{key}-al2-generic", f"{expanded_version}-al2-generic", f"{key}-al2-generic-jdk"]
     al2023_tags = [f"{key}-al2023",  f"{expanded_version}-al2023" ,f"{key}-al2023-jdk"]
+    al2023_generic_tags = [f"{key}-al2023-generic", f"{expanded_version}-al2023-generic", f"{key}-al2023-generic-jdk"]
```
Comment (lutkerd) [inline bin/tag-generator.py:15]: Can this be in the `if key in AL2023_GENERIC_VERSIONS:`?

Comment (rgithubli) [reply bin/tag-generator.py:15]: It surely can from python perspective. However, would it easier to read with all tags list grouped together on the top (along with `al2_tags` and `al2023_tags`)?

**Review by lutkerd — APPROVED** [review]

**Comment by rgithubli** [comment]
Hmm broke a  https://github.com/corretto/corretto-docker/actions/runs/5743626153/job/15568296604#logs. Will fix in another pr.

**Pull request merged**